GlimpseImputationPipeline/Glimpse2SplitReference.wdl

```diff
@@ -0,0 +1,134 @@
+        String contig_name_in_reference_panel = contig_names_in_reference_panel[i_contig]
+        String contig_name_in_genetic_maps = contig_names_in_genetic_maps[i_contig]
+
+        String reference_filename = reference_panel_prefix + contig_name_in_reference_panel + reference_panel_suffix
+        String genetic_map_filename = genetic_map_path_prefix + contig_name_in_genetic_maps + genetic_map_path_suffix
+
+        call GlimpseSplitReferenceTask {
+            input:
+                reference_panel = reference_filename,
+                reference_panel_index = reference_filename + reference_panel_index_suffix,
+                contig = contig_region,
+                i_contig = i_contig,
+                genetic_map = genetic_map_filename,
+                seed = seed,
+                min_window_cm = min_window_cm,
+                preemptible = preemptible,
+                docker = docker,
+                monitoring_script = monitoring_script
+        }
+    }
+
+    output {
+        Array[File] chunks = flatten(GlimpseSplitReferenceTask.chunks)
+        Array[File] reference_chunks = flatten(GlimpseSplitReferenceTask.split_reference_chunks)
+        Array[File?] split_reference_monitoring = GlimpseSplitReferenceTask.monitoring
+    }
+}
+
+task GlimpseSplitReferenceTask {
+    input {
+        String contig
+        Int i_contig
+        File reference_panel
+        File reference_panel_index
+        File genetic_map
+
+        Int? seed
+        Int? min_window_cm
+
+        Int mem_gb = 4
+        Int cpu = 4
+        Int disk_size_gb = ceil(2.2 * size(reference_panel, "GiB") + size(genetic_map, "GiB") + 100)
+        Int preemptible = 1
+        String docker
+        File? monitoring_script
+    }
+
+    String reference_output_dir = "reference_output_dir"
+
+    command <<<
+        set -xeuo pipefail
+
+        ~{"bash " + monitoring_script + " > monitoring.log &"}
+
+        # Print chunk index to variable
+        CONTIGINDEX=$(printf "%04d" ~{i_contig})
+
+        /GLIMPSE/GLIMPSE2_chunk --input ~{reference_panel} --region ~{contig} --map ~{genetic_map} --sequential --threads ~{cpu} --output chunks_contigindex_${CONTIGINDEX}.txt ~{"--seed "+seed} ~{"--window-cm "+min_window_cm} 
+
+        mkdir -p ~{reference_output_dir}
+
+        I_CHUNK=0
+        while IFS="" read -r LINE || [ -n "$LINE" ];
+        do
+            # Extract coordinates from chunks.txt file
+            printf -v ID "%02d" $(echo $LINE | cut -d" " -f1)
+            IRG=$(echo $LINE | cut -d" " -f3)
+            ORG=$(echo $LINE | cut -d" " -f4)
+
+            # Print chunk index to variable
+            CHUNKINDEX=$(printf "%04d" $I_CHUNK)
+
+            /GLIMPSE/GLIMPSE2_split_reference --reference ~{reference_panel} --map ~{genetic_map} --input-region ${IRG} --output-region ${ORG} --output ~{reference_output_dir}/reference_panel_contigindex_${CONTIGINDEX}_chunkindex_${CHUNKINDEX} ~{"--seed "+seed}
```

Comment:
how long does this `split_reference` command generally take?  Does it make more sense to separate `chunk` and `split_reference` into two tasks, and scatter the split_reference task over the chunks, instead of putting it in a while loop?

Reply:
For chr1, `chunk` takes about 5 minutes and `split_reference` takes about 2-3 minutes each, or about an hour total. Scattering over each chunk will definitely make things a little faster, although we'd have to localize the reference panel a couple hundred times. I would personally leave it like this and tolerate an hour runtime, but happy to change it, it's a very quick change